Merge IDForm into IDPage

erikphansen opened the pull request:
## Description
Kind of an experiment to see how things would look without making `IDForm` its own component and preventing the prop-drilling that that arrangement requires.

Also implemented a sort of prefill behavior to transfer data from the ID Form to the actual HCA so the user doesn't have to re-enter their name, DOB, and SSN.

## Testing done
Local

## Acceptance criteria
- [ ]

## Definition of done
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [ ] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

erikphansen committed:
Merge IDForm into IDPage

erikphansen committed:
Refactor IDPage component with more declarative props

rianfowler commented on src/applications/hca/containers/IDPage.jsx:
```diff
@@ -49,12 +63,73 @@ class IDPage extends React.Component {
+
+  renderLoginRequiredAlert = () => (
+    <>
+      <AlertBox
```
You could move them into their own stateless functional components. They don't have to be separate files and you could reference them as something like `loginRequired && <LoginRequiredAlert />`.

rianfowler commented on src/applications/hca/containers/IDPage.jsx:
```diff
@@ -104,15 +190,26 @@ const mapDispatchToProps = {
+    shouldRedirect: isLoggedIn(state),
+    showContinueButton: !loginRequired,
+    showLoadingIndicator: isProfileLoading(state),
+    showServerError: hasServerError,
```
+1

rianfowler replied:
Looks like there might be some `ownProps` passed into this that could be documented in a `propTypes` object. I've also seen `ownProps` plucked out in `mapStateToProps` so props from state and own props are assigned in the same place- not sure how that works exactly.

erikphansen replied:
Will add `propTypes`. Do you have any thoughts on documenting the props that come from `mapStateToProps` in `propTypes` as well? It looks like something we haven't really done much in the past, but I know it's doable.

rianfowler replied:
Yes. The propTypes list should include every prop. 

The way I see it, mapStateToProps is part of the react-redux connect component- not the component it wraps. It’s a helper that takes the state and transforms it to the propsTypwa of the component and handles componentShouldUpdate for you automatically.

rianfowler reviewed (approved):
looks good

erikphansen committed:
Work in PR review feedback

- Switch to stateless functional components
- Add PropTypes
- Prefill HCA with info from ID Page

erikphansen committed:
update ID Form schema to match HCA schema

erikphansen committed:
revert module resolver change

erikphansen committed:
remove errant comman from babelrc json
